capstone/src/main/webapp/script.js

```diff
@@ -31,14 +32,160 @@ async function submitInput() {
+async function handleInput(text, lang, requestedAttributes, tokenizer) {
+  // Draw the separating line for the output
+  const separator = document.getElementById('separator-container');
+  separator.innerHTML = '';
+  separator.appendChild(document.createElement('hr')); 
+	
+  // Make Perspective call for the entire submission and load graph data
+  const toxicityData = await callPerspective(text, lang, requestedAttributes);
+  loadChartsApi(toxicityData);
+
+  // Get detailed analysis if requested
+  if (tokenizer != "") {
+    getAnalysis(text, lang, requestedAttributes, tokenizer); 
+  }
+}
+
+/** Prints detailed analysis of text by word or sentence */
+async function getAnalysis(text, lang, requestedAttributes, tokenizer) {
+  // Set up the detailed analysis section
+  const analysisContainer = document.getElementById('analysis-container');
+  analysisContainer.appendChild(createAnyElement('b', 'Detailed Anaylsis'));
+  const loadingEl = document.createElement('p');
+  loadingEl.className = 'spinner-border';
+  analysisContainer.appendChild(loadingEl);
+  const result = document.createElement('p');
+  result.className = 'detailed-analysis';
+  
+  // Generate the results for every substring of the input text
+  const substrings = getSubstrings(text, tokenizer);
+  for (i = 0; i < substrings.length; i++) {
+    if (substrings[i] != '') {
+      // Get the Perspective scores for the substring & sort them descending
+      const response = await callPerspective(substrings[i], lang, requestedAttributes);
+      if (typeof(response.error) != 'undefined') {
+        analysisContainer.removeChild(loadingEl);
+        analysisContainer.appendChild(createAnyElement('p', 'Perspective API was not able to get scores for detailed analysis'));
+        return;
+      }
+      const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
+      const attributes = sortAttributes(response);
+
+      // Color the substring appropriately	
+      const substringEl = createAnyElement('span', substrings[i]);
+      colorSubstring(substringEl, toxicityScore);
+     
+      // Attach a tooltip (info-box for the substring)
+      const tooltipEl = createTooltip(attributes);
+      substringEl.appendChild(tooltipEl);
+      result.appendChild(substringEl);
+    }
+  }
+  analysisContainer.removeChild(loadingEl);
+  analysisContainer.appendChild(result);
+}
+
+/** Sorts a Perspective API response's attribute values in descending order */
+function sortAttributes(response) { 
+  const attributes = [];
+  Object.keys(response.attributeScores).forEach((attribute) => {
+    attributes.push([attribute, response.attributeScores[attribute].summaryScore.value]);
+  });
+  attributes.sort(function(a, b) {
+    return b[1] - a[1];
+  });
+  return attributes;
+}
+
+/** Breaks up a string into its words or sentences and puts the substrings in an array */
+function getSubstrings(text, tokenizer) {
```

Comment:
Since you only support 2 options for tokenizer here, do you want to make this some boolean instead?  Or if you expect to have more values in the future you could do some enumerated value, maybe like this: https://stijndewitt.com/2014/01/26/enums-in-javascript/

Reply:
I chose an enumerated value `TokenizerEnum`